### PR TITLE
 backupccl: fix flake in TestProtectedTimestampDuringBackup 

### LINKED ISSUE #47522: backupccl: TestProtectedTimestampsDuringBackup is flaky on master
On 82294d100ad1cb26b488f53e7241242d474d58d7, I ran
```
roachprod create lucy-stress
make roachprod-stress CLUSTER=lucy-stress STRESSFLAGS='-p 16' PKG=./pkg/ccl/backupccl TESTS=TestProtectedTimestampsDuringBackup
```
 and got this failure within a few hundred runs:
```
--- FAIL: TestProtectedTimestampsDuringBackup (57.99s)
    backup_test.go:3673: condition failed to evaluate within 45s: expected (?s)shouldQueue=true.*processing replica.*GC score after GC in trace:      0.000ms      0.000ms    === operation:manual gc queue run node:1 range:33/1:/{Table/52-Max} store:1 unfinished:
             0.012ms      0.012ms    event:[n1,s1,r33/1:/{Table/52-Max}] running gc.shouldQueue
             0.038ms      0.026ms    event:[n1,s1,r33/1:/{Table/52-Max}] shouldQueue=false, priority=0.000318


        goroutine 45 [running]:
        runtime/debug.Stack(0xc0041d3c60, 0x481bc60, 0xc002b13930)
                /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4913400, 0xc0005c5100, 0xc0041d3c60)
                /go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:37 +0x6b
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test.TestProtectedTimestampsDuringBackup(0xc0005c5100)
                /go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:3673 +0x802
        testing.tRunner(0xc0005c5100, 0x41d5550)
                /usr/local/go/src/testing/testing.go:909 +0xc9
        created by testing.(*T).Run
                /usr/local/go/src/testing/testing.go:960 +0x350
```
Full logs: [out.1586960612.gz](https://github.com/cockroachdb/cockroach/files/4481811/out.1586960612.gz)

### PR DESCRIPTION
**Pull request by pbardea**
TestProtectedTimestampDuringBackup would sometimes flake as the GC queue
would assign it a low priority that would be below the threshold to turn
shouldQueue to true. However, the priority is non-zero indicating that
the timestamp was indeed not protected. This change aims to remove the
flake by checking for a non-zero priority rather than if shouldQueue is
true.

This PR also makes the same change to the ImportInto variant of the test since
they share the same test structure.

Fixes #47522.

Release note: None

### CODE REVIEW

File: pkg/ccl/importccl/import_into_test.go [hunk 1]
```diff
@@ -147,11 +147,14 @@ func TestProtectedTimestampsDuringImportInto(t *testing.T) {
 
 	// Wait for the ranges to learn about the removed record and ensure that we
 	// can GC from the range soon.
-	gcRanRE := regexp.MustCompile("(?s)shouldQueue=true.*processing replica.*GC score after GC")
+	// This regex matches when all float priorities other than 0.00000.
+	matchNonZeroSmall := "0\\.\\d*[1-9]"       // matches 0.00123, 0.00001
+	matchNonZeroLarge := "[1-9]\\d*(\\.\\d+)?" // matches 123.000, 123.2, and 1234
+	nonZeroProgressRE := regexp.MustCompile(fmt.Sprintf("priority=(%s|%s)", matchNonZeroSmall, matchNonZeroLarge))
```
Comment (pbardea) [inline pkg/ccl/importccl/import_into_test.go:153]: I wanted to simplify this, but this was the best I could come up with given that Go's regex doesn't support negative lookahead... if any ideas come to mind, I'm open to suggestions. :)

File: pkg/ccl/backupccl/backup_test.go [hunk 2]
```diff
@@ -3669,11 +3676,14 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
 
 	// Wait for the ranges to learn about the removed record and ensure that we
 	// can GC from the range soon.
-	gcRanRE := regexp.MustCompile("(?s)shouldQueue=true.*processing replica.*GC score after GC")
+	// This regex matches when all float priorities other than 0.00000.
```
Comment (ajwerner) [inline pkg/ccl/backupccl/backup_test.go:3679]: how about: `[1-9]\d*\.\d+|0\.\d*[1-9]\d*`

**Comment by pbardea** [comment]
Updated the regex - RFAL

**Review by ajwerner — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, 1 of 1 files at r5, 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47582#-:-M5S9bpB95OuX3MiTGFr:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by pbardea** [comment]
TFTR!
bors r+

**Pull request merged**